src/Primary_Extension/org/lobobrowser/primary/ext/ComponentSource.java

```diff
@@ -682,9 +683,19 @@ public void go() {
   }
 
   public void navigateOrSearch() {
-    final String addressText = this.addressField.getText();
+    final String addressText = this.addressField.getText().trim();
+    final int periodIdx = addressText.indexOf('.');
+    final int spaceIdx = addressText.indexOf(' ');
+    final int aboutIdx = addressText.indexOf("about:");
     if (addressText.charAt(0) == '?') {
       this.search();
+    } else if ((spaceIdx != -1 || periodIdx == -1) && aboutIdx == -1) {
+      try {
+        URL url = new URL("about:confirmSearch?" + addressText);
```

Comment:
minor: we use `final` for declarations wherever possible. Can you add it here? It increases verbosity but makes it easier to understand the intent and catches bugs early on.